packages/zoe/src/contracts/multipoolAutoswap/multipoolAutoswap.js

```diff
@@ -0,0 +1,91 @@
+// @ts-check
+
+import { assert, details } from '@agoric/assert';
+import makeWeakStore from '@agoric/weak-store';
+
+import { assertIssuerKeywords } from '../../contractSupport';
+import { makeAddPool } from './pool';
+import { makeGetCurrentPrice } from './getCurrentPrice';
+import { makeMakeSwapInvitation } from './swap';
+import { makeMakeAddLiquidityInvitation } from './addLiquidity';
+import { makeMakeRemoveLiquidityInvitation } from './removeLiquidity';
+
+import '../../../exported';
+
+/**
+ * Autoswap is a rewrite of Uniswap. Please see the documentation for more
+ * https://agoric.com/documentation/zoe/guide/contracts/autoswap.html
+ *
+ * We expect that this contract will have tens to hundreds of issuers.
+ * Each liquidity pool is between the central token and a secondary
+ * token. Secondary tokens can be exchanged with each other, but only
+ * through the central token. For example, if X and Y are two token
+ * types and C is the central token, a swap giving X and wanting Y
+ * would first use the pool (X, C) then the pool (Y, C). There are no
+ * liquidity pools between two secondary tokens.
+ *
+ * There should only need to be one instance of this contract, so liquidity can
+ * be shared as much as possible.
+ *
+ * When the contract is instantiated, the central token is specified in the
+ * issuerKeywordRecord. The party that calls `E(zoe).startInstance` gets an invitation
+ * that can be used to request an invitation to add liquidity. The same
+ * invitation is available by calling `await E(publicAPI).getLiquidityIssuer(brand)`.
```

Comment:
What is the significance of the `await` here? An `await` is not something one calls.